subprojects/core/src/integTest/groovy/org/gradle/api/internal/project/ProjectIdentityIntegrationTest.groovy

```diff
@@ -82,6 +82,7 @@ class ProjectIdentityIntegrationTest extends AbstractIntegrationSpec {
         given:
         executer.requireIsolatedDaemons()
         settingsFile """
+            System.gc() // On re-run, encourage collection of the previous project identity
```

Comment:
❓Can you explain why we need this? The test is `project identity doesn't survive build finish`. So it should be the case for the Gradle without additional GC triggering. If the embedded executer makes us to do that, then we should run this test only with forking one I think

Reply:
The behavior is dependent on the `WeakHashMap` being cleared after the build finish. Without the GC call, it may not happen and therefore the same identity will be seen on the next run.

I imagine this was originally working because each build created a new project that had a different object identity (i.e., this test didn't really test anything before, it just put two entries in the map). However, we now use the identity path, so we need to make sure that the old `Project` is actually cleaned up.

Reply:
> I imagine this was originally working because each build created a new project that had a different object identity (i.e., this test didn't really test anything before, it just put two entries in the map).

Right. Honestly, I don't see any value in this test with the current implementation anymore

Reply:
Thinking over it more, I think it does have value with the `IdentityHashMap`. It ensures that we don't keep project instances across builds. I removed the other test because I don't think that one is useful, and it doesn't work under IP anyways as we purposefully don't keep the same project object between subprojects, we wrap.